pricegraph/src/orderbook.rs

```diff
@@ -355,23 +356,17 @@ impl Orderbook {
 
             // NOTE: `capacity` is expressed in the buy token, so we need to
             // divide by the exchange rate to get the sell amount being filled.
-            let fill_amount = flow.capacity / transitive_xrate.value();
-
-            order.amount -= fill_amount;
-            debug_assert!(
-                order.amount >= -num::max_rounding_error(fill_amount),
-                "remaining amount underflow for order {}-{}",
-                order.user,
-                order.id,
-            );
-
+            let fill_amount = (flow.capacity / transitive_xrate.value()) as u128;
             let new_balance = user.deduct_from_balance(pair.sell, fill_amount);
 
             if num::is_dust_amount(new_balance) {
                 user.clear_balance(pair.sell);
                 self.update_projection_graph_node(pair.sell);
-            } else if num::is_dust_amount(order.amount) {
-                self.update_projection_graph_edge(pair);
+            } else if let Some(amount) = &mut order.amount {
+                *amount = amount.saturating_sub(fill_amount);
```

Comment:
nit: Can we re-introduce the `debug_assert` check on `amount -= fill_amount` to make sure that we aren't over-filling orders? I guess that right now this is hard because some of this is in floats and rounding errors can cause this to fail. Maybe we can leave a `TODO` to put the assertion back in once we switch to rational numbers?